### PR TITLE
ENH: Add documentation on how to develop the extension.

### PR DESCRIPTION
**Pull request by thewtex**
@mlp6 Please take a look. I have to create user documentation for the modules, then we can add it to the Slicer Extension manager.

### CODE REVIEW

**Comment by mlp6** [comment]
Will do!

File: README.rst [hunk 1]
```diff
@@ -6,3 +6,25 @@ SlicerITKUltrasound
...
+*<Slicer-Superbuild>/Slicer-build* path. Build SlicerITKUltrasound by calling
+`make` or the other specific build system generated by CMake.
+
+Run the tests for the extension by running `ctest` from
+*<SlicerITKUltrasound-Superbuild>/SlicerITKUltrasound-build*.
+
+To use the modules from Slicer's user interface: add the generated modules to
+Slicer's module search path by opening Slicer, then *Edit -> Application
+Settings -> Modules -> Additional module paths -> Add
+<SlicerITKUltrasound-Superbuild>/SlicerITKUltrasound-build/lib/Slicer-4.N/cli-modules*,
```
Comment (jcfr) [inline README.rst]: Nice 

Minor nitpick:

`lib/Slicer-4.N` -> `lib/Slicer-X.Y`

We use this convention in Slicer documentation. See http://wiki.slicer.org/slicerWiki/index.php/Documentation/Nightly/Developers/DirectoryStructure

Comment (thewtex) [reply README.rst]: @jcfr Thanks for the review! Addressing in follow-up patch.

File: README.rst [hunk 2]
```diff
@@ -6,3 +6,25 @@ SlicerITKUltrasound
...
+<https://www.slicer.org/slicerWiki/index.php/Documentation/Nightly/Developers/Build_Instructions>`_.
+Version *r25484* (trunk from 2016-10-28) or later is required.
+
+Use CMake to configure SlicerITKUltrasound, and `pass in
+<https://www.slicer.org/slicerWiki/index.php/Documentation/Nightly/Developers/Build_Module>`_
+the `Slicer_DIR` configuration variable as the
+*<Slicer-Superbuild>/Slicer-build* path. Build SlicerITKUltrasound by running
+`make` or another specific build system generated by CMake.
+
+Run the tests for the extension by executing `ctest` in
```
Comment (jcfr) [inline README.rst:23]: If building on windows, `ctest -C Release` or `ctest -C Debug`

Comment (mlp6) [reply README.rst:23]: Windows builds are definitely on the radar.

Comment (thewtex) [reply README.rst:23]: I haven't tested on Windows yet, but it "should work". Adding `-C` suggestion to the docs.

**Review by jcfr — APPROVED** [review]

**Comment by thewtex** [comment]
@jcfr Thanks for the review.

Merging as is with improvements and fixes expected in the future.

@mlp6 please report an issues encountered in the issue tracker.

**Pull request merged**